### DISCUSSION TITLE
shapely.transform vs shapely.ops.transform

### DISCUSSION THREAD

**Discussion by ijessen-mitll** [shapely/shapely#1895] [Q&A] [answered]
The two `transform` functions seem to significantly overlap each other:

> shapely.ops.transform(func, geom)
> Applies func to all coordinates of geom and returns a new geometry of the same type from the transformed coordinates.

> shapely.transform(geometry, transformation, include_z=False)
> Returns a copy of a geometry array with a function applied to its coordinates.

There are very slight differences in their parameters, and some logic within `ops.transform` seems to try to suss out the interface of the provided transform function.  Besides that, can anyone speak to why both exist?  What are the independent use cases for each?  Can the documentation be improved to direct the use of one over the other?

**Comment by mwtoews** [comment, marked as answer]
The quick answer is that [`shapely.ops.transform`](https://shapely.readthedocs.io/en/1.8.5.post1/manual.html#shapely.ops.transform) has been with shapely for many versions and `shapely.transform` is newer, since version 2.0. It was merged from pygeos' [`apply`](https://pygeos.readthedocs.io/en/stable/coordinates.html#pygeos.coordinates.apply) with the same arguments.
